docs/public/using/backend/rpcapi-reference.md

```diff
@@ -1436,6 +1439,149 @@ If the `batch_id` is undefined the following error is returned:
 }
 ```
 
+
+### API method: `batch_status`
+
+This method replaces deprecated method
+[API method: get_batch_job_result](#api-method-get_batch_job_result).
+
+Returns the number of waiting, running and finished *tests*. Optionally it also
+returns the [*test ids*][Test id] of the *batch test*, in three different lists
+(waiting, running, finished). Only non-empty lists are included.
+
+Example valid request and response, respectively, where no [*test ids*][Test id]
+were requested:
+
+```json
+{
+    "jsonrpc": "2.0",
+    "id": 147559211994909,
+    "method": "batch_status",
+    "params": {
+        "batch_id": "8"
+    }
+}
+```
+
+```json
+{
+   "jsonrpc": "2.0",
+   "id": 147559211994909,
+   "result": {
+      "finished_count": 5,
+      "running_count": 5,
+      "waiting_count": 195
+   }
+}
+```
+
+Example valid request  where [*test ids*][Test id] were requested for
+all three status values:
+
+```json
+{
+    "jsonrpc": "2.0",
+    "id": 147559211994909,
+    "method": "batch_status",
+    "params": {
+        "batch_id": "8",
+        "list_running_tests": true,
+        "list_finished_tests": true,
+        "list_waiting_tests": true
+    }
+}
+```
+
+Example valid response where finished [*test id*][Test id] objects were
+requested but none existed:
+
+```json
+{
+   "jsonrpc": "2.0",
+   "id": 147559211994909,
+   "result": {
+      "finished_count": 0,
+      "running_count": 5,
+      "waiting_count": 195
+   }
+}
+```
```

Comment:
From what I understand, this is a response to a request where `list_finished_tests` is `true`, `list_waiting_tests` is `false` and `list_running_tests` is `false`.

I’d expect to always have a `finished_tests` key in the response if and only if `list_finished_tests` is `true`, even if the list is empty.